CONTRIBUTING.md

```diff
@@ -65,6 +90,170 @@ can check the XML source out from Subversion and edit that and build it
+  changed. It is important for the stability and maintainability of PHP that
+  tests are comprehensive.
+* For development use the `--enable-debug` switch to avoid memory leaks and the
+  `--enable-maintainer-zts` switch to ensure your code handles TSRM correctly and
+  doesn't break for those who need that. Check the PHP and web server error logs
+  after running your PHP tests.
+* Update Git source from upstream before testing and submitting your patch.
+* Review the patch once more just before submitting it.
+
+The next few rules are more of a technical nature:
+
+* Follow [CODING_STANDARDS](/CODING_STANDARDS) before you start working.
+* All changes should first go to the lowest supported branch (i.e. 7.0) and then
+  get merged up to all other branches. If a change is not needed for later branches
+  (i.e. fixes for features which were dropped from later branches) an empty merge
+  should be done.
+* All news updates intended for public viewing, such as new features, bug fixes,
+  improvements, etc., should go into the NEWS file of *any stable release* version
+  with the given change. In other words, news about a bug fix which went into
+  PHP-7.1, PHP-7.2 and master should be noted in PHP-7.1/NEWS and PHP-7.2/NEWS
+  but not master, which is not a public released version yet.
+* Do not commit multiple files and dump all messages in one commit. If you
+  modified several unrelated files, commit each group separately and provide a
+  nice commit message for each one. See example below.
+* Do write your commit message in such a way that it makes sense even without the
+  corresponding diff. One should be able to look at it, and immediately know what
+  was modified. Definitely include the function name in the message as shown below.
+* In your commit messages, keep each line shorter than 80 characters. And try to
+  align your lines vertically, if they wrap. It looks bad otherwise.
+* If you modified a function that is callable from PHP, prepend PHP to the
+  function name.
+* Add in-line comments and/or have external documentation ready.
+
+The format of the commit messages is pretty simple.
+
+```
+<max 79 characters short description>\n
+\n
+<long description, 79 chars per line>
+\n
+```
+
+An example from the git project (commit 2b34e486bc):
+
+```
+pack-objects: Fix compilation with NO_PTHREDS
+
+It looks like commit 99fb6e04 (pack-objects: convert to use
+parse_options(), 2012-02-01) moved the #ifdef NO_PTHREDS around but
+hasn't noticed that the 'arg' variable no longer is available.
+```
+
+If you fix some bugs, you should note the bug ID numbers in your commit message.
+Bug ID should be prefixed by `#` for easier access to bug report when developers
+are browsing VCS via LXR or Bonsai.
+
+Example:
+
+```
+Fixed bug #14016 (pgsql notice handler double free crash bug.)
+```
+
+When you change the NEWS file for a bug fix, then please keep the bugs sorted in
+decreasing order under the fixed version.
```

Comment:
Unless you are going to commit it yourself, you shouldn't include the NEWS entry as it will probably create conflicts for the person merging it.

Reply:
Added.